### PR TITLE
feat: update kkcore CRDs

### PR DESCRIPTION
**Pull request by dbbDylan**
### What type of PR is this?

/kind feature
/kind design

### What this PR does / why we need it:
Update kkcore CRDs for CAPKK

### Which issue(s) this PR fixes:

Fixes #

### Does this PR introduced a user-facing change?

```release-note
- update: add `PipelineTemplate` CRD for CAPKK, used for create `Pipeline`.
- update: add `.Status.HostMachineMapping` for `Inventory`, used for update status between `KKMachine` and `Hosts`.
```

### CODE REVIEW

File: pkg/apis/core/v1/inventory_types.go [hunk 1]
```diff
@@ -66,6 +80,33 @@ type InventoryList struct {
...
+					unavailableHosts[hostName] = struct{}{}
+					validHosts = append(validHosts, hostName)
+				}
+			}
+		}
+		hosts = append(hosts, validHosts...)
+	}
+
+	return hosts
+}
```
Comment (redscholar) [inline pkg/apis/core/v1/inventory_types.go:108]: move in variable package

File: pkg/apis/core/v1/pipelinetemplate_types.go
Comment (redscholar) [inline pkg/apis/core/v1/pipelinetemplate_types.go:1]: should not set as a crd.

File: pkg/apis/core/v1/inventory_types.go (hunk 1 above)
Comment (dbbDylan) [reply pkg/apis/core/v1/inventory_types.go:108]: This logic is unique for `inventory` CRD, so I don't need to move it.

By the way, `variable` is for `playbook`, which will store variables when the playbook being executed. But `GetHostsFromGroup` used for init a valid `inventory`.

File: pkg/apis/core/v1/pipelinetemplate_types.go
Comment (dbbDylan) [reply pkg/apis/core/v1/pipelinetemplate_types.go:1]: Has been justified in PR[#2425 ](https://github.com/kubesphere/kubekey/pull/2425)

**Pull request merged**